src/index.js

```diff
@@ -0,0 +1,78 @@
+import readline from "readline";
+
+const RANDOM_NUMBER_RANGE = 50
+const ANSWER_LIMIT = 6
+const ARR_LAST_IDX = -1
+
+function readLineAsync(query) {
+    // 내부 로직 구현
+     // readline 인터페이스 생성
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  // 사용자 입력을 Promise로 반환
+  return new Promise((resolve) => {
+    rl.question(query, (input) => {
+      resolve(input);
+      rl.close();
+    });
+  });
+}
+
+async function reset(callback) {
+  const resetGame = await readLineAsync("게임을 다시 시작하시겠습니까? (yes/no): ");
+  if(resetGame != "yes" && resetGame != "no") {
```

Comment:
P2;

가독성을 위해 space를 넣으면 좋을 것 같아요.
```suggestion
  if (resetGame != "yes" && resetGame != "no") {
```

Reply:
ask;

얕은 비교와 깊은 비교 둘의 차이에 대해 알고 계신가요!?
저는 개인적으로 정말 특수한 상황이 아니면 깊은 비교를 주로 활용하고 있어요.

해당 코드들 또한 깊은 비교로 처리하면 좋을 것 같은데 어떠신가요?

Reply:
이부분도 확인해서 깊은 비교로 처리했습니다!